crates/polars-core/src/frame/mod.rs

```diff
@@ -411,6 +409,28 @@ impl DataFrame {
         DataFrame { columns }
     }
 
+    /// Create a new `DataFrame` but does not check the length of the `Series`,
+    /// only check for duplicates.
+    ///
+    /// It is advised to use [DataFrame::new](DataFrame::new) in favor of this method.
+    ///
+    /// # Panic
+    /// It is the callers responsibility to uphold the contract of all `Series`
+    /// having an equal length, if not this may panic down the line.
+    pub fn new_no_length_checks(columns: Vec<Series>) -> PolarsResult<DataFrame> {
```

Comment:
This should be tagged `unsafe` as it doesn't check the invariant of `DataFrame`.